### PR TITLE
Icons: rename story-pin icon to idea-pin + codemod

### PR DESCRIPTION
**Pull request by AlbertCarreras**
### Summary
Icons: rename story-pin icon to idea-pin

### Links

- [Jira](link to Jira ticket(s))
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        !localModuleIdentifierName.includes(node.openingElement.name.name)
+      ) {
+        return null;
+      }
+
+      const attrs = node.openingElement.attributes;
+
+      if (attrs.some((attr) => attr.type === 'JSXSpreadAttribute')) {
+        throw new Error(
+          `Remove Dynamic Text properties and rerun codemod. Location: ${file.path} @line: ${node.loc.start.line}`,
```
Comment (dangerismycat) [inline packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js:58]: I don't think we're dealing with Text here 😅

File: packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ * Converts
+ *  <Icon icon='story-pin'/> to <Icon icon='idea-pin'/>
+ *  <IconButton icon='story-pin'/> to <IconButton icon='idea-pin'/>
+ *  <Pog icon='story-pin'/> to <Pog icon='idea-pin'/>
+ *  <Button iconEnd='story-pin'/> to <Button iconEnd='idea-pin'/>
+ *
+ *  Console logs each Module for a manual review.
+ */
+
+// yarn codemod --parser=flow -t=packages/gestalt-codemods/19.0.0/rename-story-pin-icon-to-idea-pin.js relative/path/to/your/code
```
Comment (dangerismycat) [inline packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js:11]: s/19.0.0/40.0.0/

File: packages/gestalt-codemods/40.0.0/__testfixtures__/rename-story-pin-icon-to-idea-pin.input.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// @flow strict
+import { Box, Icon, IconButton, Pog, Button, Icon as RenamedIcon } from 'gestalt';
+
+export default function TestBox() {
+  return (
+    <Box>
+      <Icon icon='story-pin' size={16}/>
```
Comment (dangerismycat) [inline packages/gestalt-codemods/40.0.0/__testfixtures__/rename-story-pin-icon-to-idea-pin.input.js:7]: We should probably also cover ternaries and variables

File: packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js:58]: DONE

File: packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt-codemods/40.0.0/rename-story-pin-icon-to-idea-pin.js:11]: DONE

File: packages/gestalt-codemods/40.0.0/__testfixtures__/rename-story-pin-icon-to-idea-pin.input.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt-codemods/40.0.0/__testfixtures__/rename-story-pin-icon-to-idea-pin.input.js:7]: DONE

**Review by dangerismycat — APPROVED** [review]

**Pull request merged**